### PR TITLE
Joshua s brown/ci darwin speedup

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
## PR Summary

Speeds up the ci by about an hour by avoiding spack install from scratch and instead using the project space @AndrewGaspar setup. 

Waiting for pr https://github.com/lanl/parthenon/pull/200 to be merged first. 

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented. From the previous pr. 
- [ ] Adds a test for any bugs fixed. Adds tests for new features. Not applicable
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md

### CODE REVIEW

**Comment by JoshuaSBrown** [comment]
https://xcp-gitlab.lanl.gov/eap-oss/parthenon/-/jobs/88726

File: scripts/darwin/build_hdf5_parallel.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
```
Comment (AndrewGaspar) [inline scripts/darwin/build_hdf5_parallel.sh]: This can go away, right?

Comment (AndrewGaspar) [reply scripts/darwin/build_hdf5_parallel.sh]: I guess the answer was yes! :)

Comment (JoshuaSBrown) [reply scripts/darwin/build_hdf5_parallel.sh]: Yes :).

**Comment by JoshuaSBrown** [comment]
This should also be squashed when merged.

**Review by AndrewGaspar — APPROVED** [review]

**Review by Yurlungur — APPROVED** [review]
Should we manually trigger a long test, i.e., the full regression suite, before merge?

**Comment by JoshuaSBrown** [comment]
https://xcp-gitlab.lanl.gov/eap-oss/parthenon/-/jobs/88854

**Review by pgrete — APPROVED** [review]
Ready to squash and merge.

**Comment by JoshuaSBrown** [comment]
I'm going to go ahead and merge this @pgrete this should not affect anything other than our ci, but if you have time and want to offer feedback I would still be happy to receive it.

**Comment by JoshuaSBrown** [comment]
Wow, you literally just looked at this. Great.

**Pull request merged**